actors/builtin/market/market_test.go

```diff
@@ -352,9 +354,289 @@ func TestPublishStorageDeals(t *testing.T) {
+				},
+				exitCode: exitcode.ErrIllegalArgument,
+			},
+			"negative price per epoch": {
+				setup: func(rt *mock.Runtime, _ *marketActorTestHarness, d *market.DealProposal) {
+					d.StoragePricePerEpoch = abi.NewTokenAmount(-1)
+				},
+				exitCode: exitcode.ErrIllegalArgument,
+			},
+			"price per epoch greater than total filecoin": {
+				setup: func(rt *mock.Runtime, _ *marketActorTestHarness, d *market.DealProposal) {
+					d.StoragePricePerEpoch = big.Add(abi.TotalFilecoin, big.NewInt(1))
+				},
+				exitCode: exitcode.ErrIllegalArgument,
+			},
+			"negative provider collateral": {
+				setup: func(rt *mock.Runtime, _ *marketActorTestHarness, d *market.DealProposal) {
+					d.ProviderCollateral = big.NewInt(-1)
+				},
+				exitCode: exitcode.ErrIllegalArgument,
+			},
+			"provider collateral greater than max collateral": {
+				setup: func(rt *mock.Runtime, _ *marketActorTestHarness, d *market.DealProposal) {
+					d.ProviderCollateral = big.Add(abi.TotalFilecoin, big.NewInt(1))
+				},
+				exitCode: exitcode.ErrIllegalArgument,
+			},
+			"negative client collateral": {
+				setup: func(rt *mock.Runtime, _ *marketActorTestHarness, d *market.DealProposal) {
+					d.ClientCollateral = big.NewInt(-1)
+				},
+				exitCode: exitcode.ErrIllegalArgument,
+			},
+			"client collateral greater than max collateral": {
+				setup: func(rt *mock.Runtime, _ *marketActorTestHarness, d *market.DealProposal) {
+					d.ClientCollateral = big.Add(abi.TotalFilecoin, big.NewInt(1))
+				},
+				exitCode: exitcode.ErrIllegalArgument,
+			},
+			"client does not have enough balance for collateral": {
+				setup: func(rt *mock.Runtime, a *marketActorTestHarness, d *market.DealProposal) {
+					a.addParticipantFunds(rt, client, big.Sub(d.ClientBalanceRequirement(), big.NewInt(1)))
+					a.addProviderFunds(rt, d.ProviderCollateral)
+				},
+				exitCode: exitcode.ErrInsufficientFunds,
+			},
+			"provider does not have enough balance for collateral": {
+				setup: func(rt *mock.Runtime, a *marketActorTestHarness, d *market.DealProposal) {
+					a.addParticipantFunds(rt, client, d.ClientBalanceRequirement())
+					a.addProviderFunds(rt, big.Sub(d.ProviderCollateral, big.NewInt(1)))
+				},
+				exitCode: exitcode.ErrInsufficientFunds,
+			},
+			"unable to resolve client address": {
+				setup: func(rt *mock.Runtime, a *marketActorTestHarness, d *market.DealProposal) {
+					d.Client = tutil.NewBLSAddr(t, 1)
+				},
+				exitCode: exitcode.ErrNotFound,
+			},
+			"signature is invalid": {
+				setup: func(rt *mock.Runtime, a *marketActorTestHarness, d *market.DealProposal) {
+
+				},
+				exitCode:                   exitcode.ErrIllegalArgument,
+				signatureVerificationError: errors.New("error"),
+			},
+			"no entry for client in locked  balance table": {
+				setup: func(rt *mock.Runtime, a *marketActorTestHarness, d *market.DealProposal) {
+					a.addProviderFunds(rt, d.ProviderCollateral)
+				},
+				exitCode: exitcode.ErrIllegalState,
```

Comment:
This is unfortunate. Is it possible to get the code to return ErrInsufficientFunds here too? (same below)

Reply:
This is done.